### PR TITLE
New OnCall initialization process

### PR DESCRIPTION
**Pull request by brojd**
# What this PR does

- Add backend plugin starter
- Add backend plugin unit test step to GH workflows
- Support backend plugin build in Tilt
- On the frontend support `USE_BACKEND_PLUGIN` env var. If it's `true`, then traffic is directed to backend plugin. Otherwise it goes to data source proxy as before

Closes https://github.com/grafana/oncall-private/issues/2697

Partially covers https://github.com/grafana/oncall-private/issues/2696 

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [x] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] Added the relevant release notes label (see labels prefixed w/ `release:`). These labels dictate how your PR will
    show up in the autogenerated release notes.

### CODE REVIEW

**Review by matiasb — COMMENTED** [review]
LGTM

File: grafana-plugin/pkg/plugin/proxy.go [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+	err = SetOnCallUserHeader(onCallUser, req)
+	if err != nil {
+		log.DefaultLogger.Error("Error setting user header: %v", err)
+		return err
+	}
+
+	return nil
+}
+
+func (a *App) ProxyRequestToOnCall(w http.ResponseWriter, req *http.Request, pathPrefix string) {
```
Comment (Konstantinov-Innokentii) [inline grafana-plugin/pkg/plugin/proxy.go:123]: Have u considered using built in [reverse-proxy](https://go.dev/src/net/http/httputil/reverseproxy.go)? Here is the example how it's made in [grafana](https://github.com/grafana/grafana/blob/0c2f58bdaef495e4719c999882fc315e89f405a2/pkg/util/proxyutil/reverse_proxy.go#L27). Worth to check with plugin-platform team if it's something that they are using too.

Comment (brojd) [reply grafana-plugin/pkg/plugin/proxy.go:123]: I need to close this PR in favor to https://github.com/grafana/oncall/pull/4657/files so let's discuss there

**Comment by brojd** [comment]
Closiing in favor to https://github.com/grafana/oncall/pull/4657/files

**Pull request closed without merging**